stock_picking_type_auto_serial/models/stock_pack_operation.py

```diff
@@ -0,0 +1,60 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from openerp import api, models
+from openerp.osv import fields as osv_fields
+
+
+class StockPackOperation(models.Model):
+
+    _inherit = 'stock.pack.operation'
+
+    # pylint: disable=old-api7-method-defined
+    def _compute_lots_visible(
+            self, cr, uid, ids, field_name, arg, context=None):
+        res = super(StockPackOperation, self)._compute_lots_visible(
+            cr, uid, ids, field_name=field_name, arg=arg, context=context)
+
+        packs = self.browse(cr, uid, ids, context=context)
+
+        pack_with_invisible_lots = packs.filtered(
+            lambda p, r=res: r.get(p.id) and
+            p.picking_id.picking_type_id.automatic_serial_generation and
+            p.product_id.automatic_serial_generation and
+            p.location_id.usage == 'supplier')
+        for rec in pack_with_invisible_lots:
+            res[rec.id] = False
+
+        return res
+
+    # pylint: disable=attribute-deprecated
+    _columns = {
+        'lots_visible': osv_fields.function(
+            _compute_lots_visible,
+            type='boolean',
+        ),
+    }
+
+    @api.multi
+    def _prepare_pack_ops_serials(self):
+        """
+        Generate a separate serial for each unit.
+        :param vals: dictionnary containing attribute values to create a
+        stock.pack.operation record.
+        :return: [(0, 0, {qty:1, name: 'new generated serial}), (0, 0 , ...)]
+        """
+        for rec in self:
+            int_qty = int(rec.qty_done)
+            pack_ops_lots = []
+            while int_qty != 0:
+                int_qty -= 1
+                new_lot_name = self.env['ir.sequence'].next_by_code(
```

Comment:
```self.env['ir.sequence']``` could be oustide the loop.

Reply:
@benwillig nope, we need to call for each single unit

Reply:
Hi @zakiuu 
3 comments in this section:
consider making new_lot_name a separate method - this could massively widen its usefulness - i.e. non sequence based automatic generation or separate sequences for different products is just overriding one method then.

while int_qty is sufficient

however I think

```python
for i in range(rec.qty_done):
   ...
```

is cleaner.  Personal preference really, just 3 less lines of code.